### PR TITLE
ARTEMIS-3178 Fixing validation on PagingStoreImpl for parameters set

### PR DESCRIPTION
**Pull request by clebertsuconic**

### CODE REVIEW

File: artemis-server/src/test/java/org/apache/activemq/artemis/core/config/impl/ConfigurationImplTest.java [hunk 1]
```diff
@@ -1082,6 +1083,68 @@ public void testAddressSettingsPageLimitInvalidConfiguration2() throws Throwable
 
       String randomString = RandomUtil.randomString();
 
+      properties.put("addressSettings.#.expiryAddress", randomString);
+      properties.put("addressSettings.#.pageLimitMessages", "300");
+      // properties.put("addressSettings.#.pageLimitBytes", "300000"); out on purpose
+      //properties.put("addressSettings.#.pageFullMessagePolicy", "DROP"); // removing the pageFull on purpose
```
Comment (gemmellr) [inline artemis-server/src/test/java/org/apache/activemq/artemis/core/config/impl/ConfigurationImplTest.java:1089]: The commented out code seems out of place (and inconsistently indented)...why not just have the comment explain pageLimitBytes and pageFullMessagePolicy are missing to test the validation of them being missing. Which the existing comments dont even really make clear.

File: artemis-server/src/test/java/org/apache/activemq/artemis/core/config/impl/ConfigurationImplTest.java [hunk 2]
```diff
@@ -1082,6 +1083,68 @@ public void testAddressSettingsPageLimitInvalidConfiguration2() throws Throwable
...
+      ConfigurationImpl configuration = new ConfigurationImpl();
+
+      Properties properties = new Properties();
+
+      String randomString = RandomUtil.randomString();
+
+      properties.put("addressSettings.#.expiryAddress", randomString);
+      //properties.put("addressSettings.#.pageLimitMessages", "300");
+      properties.put("addressSettings.#.pageLimitBytes", "300000"); // removing this on purpose
+      //properties.put("addressSettings.#.pageFullMessagePolicy", "DROP"); // removing the pageFull on purpose
```
Comment (gemmellr) [inline artemis-server/src/test/java/org/apache/activemq/artemis/core/config/impl/ConfigurationImplTest.java:1120]: As before, just comment about pageLimitMessages and pageFullMessagePolicy being missing to test that, without the commented-out code.

The "// removing this on purpose" comment doesnt actually match up, with it not being removed.

File: artemis-server/src/test/java/org/apache/activemq/artemis/core/config/impl/ConfigurationImplTest.java (hunk 1 above)
Comment (clebertsuconic) [reply artemis-server/src/test/java/org/apache/activemq/artemis/core/config/impl/ConfigurationImplTest.java:1089]: to me, the commented out code makes it clear on the intention.. but fine, I will remove it.

File: artemis-server/src/test/java/org/apache/activemq/artemis/core/config/impl/ConfigurationImplTest.java [hunk 3]
```diff
@@ -1082,10 +1083,69 @@ public void testAddressSettingsPageLimitInvalidConfiguration2() throws Throwable
...
+      runAfter(AssertionLoggerHandler::stopCapture);
+      ConfigurationImpl configuration = new ConfigurationImpl();
+
+      Properties properties = new Properties();
+
+      String randomString = RandomUtil.randomString();
+
+      // pageLimitMessages and pageFullMessagePolicy not set on purpose
+      properties.put("addressSettings.#.expiryAddress", randomString);
+      properties.put("addressSettings.#.pageLimitBytes", "300000"); // removing this on purpose
```
Comment (gemmellr) [inline artemis-server/src/test/java/org/apache/activemq/artemis/core/config/impl/ConfigurationImplTest.java:1118]: // removing this on purpose

it isnt being removed

**Pull request merged**